core/src/main/java/org/elasticsearch/gateway/TransportNodesListGatewayStartedShards.java

```diff
@@ -131,16 +141,17 @@ protected NodeGatewayStartedShards nodeOperation(NodeRequest request) {
         }
     }
 
-    private boolean canOpenIndex(ShardId shardId, IndexMetaData metaData) throws IOException {
-        // try and see if we an list unallocated
-        if (metaData == null) {
-            return false;
-        }
+    private boolean tryOpenIndex(ShardId shardId, IndexMetaData metaData) throws IOException {
```

Comment:
same comment as for Store.tryOpenIndex - I think we should just return void or throw an exception (which we catch higher up anyway)